src/entities/nestedExit/encodeCalls.ts

```diff
@@ -0,0 +1,119 @@
+import { Hex } from '../../types';
+import { WeightedEncoder } from '../encoders';
+import { ComposableStableEncoder } from '../encoders/composableStable';
+import { NestedExitCallAttributes } from './types';
+import { Relayer } from '../relayer';
+import { replaceWrapped } from '../utils/replaceWrapped';
+import { bathcRelayerLibraryAbi } from '../../abi';
+import { encodeFunctionData } from 'viem';
+
+export const encodeCalls = (
+    callsAttributes: NestedExitCallAttributes[],
+    isProportional: boolean,
+) => {
+    const encodedCalls: Hex[] = [];
+    for (const callAttributes of callsAttributes) {
+        const {
+            useNativeAssetAsWrappedAmountOut,
+            chainId,
+            sortedTokens,
+            poolId,
+            poolType,
+            kind,
+            sender,
+            recipient,
+            bptAmountIn,
+            minAmountsOut,
+            toInternalBalance,
+            outputReferenceKeys,
+            tokenOutIndex,
+        } = callAttributes;
+
+        // replace wrapped token with native asset if needed
+        let tokensOut = [...sortedTokens];
+        if (useNativeAssetAsWrappedAmountOut) {
+            tokensOut = replaceWrapped([...sortedTokens], chainId);
+        }
+
+        const _bptAmountIn = bptAmountIn.isRef
+            ? Relayer.toChainedReference(bptAmountIn.amount)
+            : bptAmountIn.amount;
+
+        let userData: Hex;
+        let outputReferences: { index: bigint; key: bigint }[] = [];
+
+        if (isProportional) {
+            switch (poolType) {
+                case 'Weighted':
+                    userData = WeightedEncoder.exitProportional(_bptAmountIn);
+                    break;
+                case 'ComposableStable':
+                    userData =
+                        ComposableStableEncoder.exitProportional(_bptAmountIn);
+                    break;
+                default:
+                    throw new Error('Unsupported pool type');
+            }
+
+            outputReferences = outputReferenceKeys.map((k) => {
+                const tokenIndex = k % 10n;
```

Comment:
Could you explain what this is for?

Reply:
I set outputReferenceKey in a way that its last digit represented the index of the token in the pool, so I was reusing that here to get the index out, but I agree it's non-ideal to have this kind of hidden logic in the code. I'll go ahead and update outputReferences so this info is more explicit and doesn't cause confusion.

Reply:
I did see it was used in `getSender`?

Reply:
Yeah, I added this comment before implementing `getSender`. It does help on that logic, so I'd say we should keep it 👍